rust/src/types/stack.rs

```diff
@@ -22,23 +115,30 @@ impl Stack {
         self.check_underflow(nth + 1)?;
 
         let len = self.0.len();
-        self.0.swap(len - 1, len - 1 - nth);
+        self.0.as_mut_slice().swap(len - 1, len - 1 - nth);
         Ok(())
     }
 
     pub fn pop<const N: usize>(&mut self) -> Result<[u256; N], FailStatus> {
         self.check_underflow(N)?;
 
-        let new_len = self.0.len() - N;
-        let mut array = [u256::ZERO; N];
-        array.copy_from_slice(&self.0[new_len..]);
-        self.0.truncate(new_len);
-        Ok(array)
+        #[cfg(feature = "stack-array")]
+        {
+            Ok(self.0.pop_array().unwrap())
```

Comment:
As mentioned before, there is a duplicated bound check here.
Maybe is easier to write: `return self.0.pop_array()` return a result and do the bounds check in the alternative implementation. 
same for Push